qemu/tests/mq_change_qnum.py

```diff
@@ -29,30 +29,36 @@ def change_queues_number(session, ifname, q_number, queues_status=None):
         """
         Change queues number
         """
-        mq_set_cmd = "ethtool -L %s combined %d" % (ifname, q_number)
         if not queues_status:
             queues_status = get_queues_status(session, ifname)
 
-        if (q_number != queues_status[1] and q_number <= queues_status[0] and
-                q_number > 0):
-            expect_status = 0
-        else:
-            expect_status = 1
-
+        mq_set_cmd = "ethtool -L %s combined %d" % (ifname, q_number)
         status, output = session.cmd_status_output(mq_set_cmd, safe=True)
         cur_queues_status = get_queues_status(session, ifname)
-        if status != expect_status:
-            err_msg = "Change queues number failed, "
-            err_msg += "current queues set is %s, " % queues_status[1]
-            err_msg += "max allow queues set is %s, " % queues_status[0]
-            err_msg += "when run cmd: '%s', " % mq_set_cmd
-            err_msg += "expect exit status is: %s, " % expect_status
-            err_msg += "output: '%s'" % output
-            test.fail(err_msg)
-        if not status and cur_queues_status == queues_status:
-            test.fail("params is right, but change queues failed")
-        elif status and cur_queues_status != queues_status:
-            test.fail("No need change queues number")
+
+        if (q_number != queues_status[1] and q_number <= queues_status[0]
+                and q_number > 0):
+            if (cur_queues_status[1] != q_number
+                    or cur_queues_status[0] != queues_status[0]):
+                err_msg = "Change queues number failed, "
+                err_msg += "current queues set is %s, " % queues_status[1]
+                err_msg += "max allow queues set is %s, " % queues_status[0]
+                err_msg += "when run cmd: '%s', " % mq_set_cmd
+                err_msg += "expect queues are %s," % q_number
+                err_msg += "expect max allow queues are %s, " % queues_status[0]
+                err_msg += "output: '%s'" % output
+                test.fail(err_msg)
+        else:
+            if cur_queues_status != queues_status:
+                err_msg = "Change queues number failed, "
+                err_msg += "current queues set is %s, " % queues_status[1]
+                err_msg += "max allow queues set is %s, " % queues_status[0]
+                err_msg += "when run cmd: '%s', " % mq_set_cmd
+                err_msg += "expect queues are %s," % queues_status[1]
+                err_msg += "expect max allow queues are %s, " % queues_status[0]
+                err_msg += "output: '%s'" % output
```

Comment:
The error message are totally same in the two conditions, that's really unnecessary to define them twice.

Reply:
@vivianQizhu Thanks for review. I also considered this, there is a bit different. As the key "expect queues" value is different. That's why I didn't share the codes. Does this make sense?

Reply:
@pezhang I don't think so, you expect `queues_status[1] == q_number` always, so your "expect queues" is always `q_number`.